plsync/mlabconfig_test.py

```diff
@@ -153,27 +158,29 @@ def test_export_experiment_records_flattened(self):
                                    use_initscript=True,
                                    ipv6='all')]
         expected_results = [
-            mlabconfig.format_a_record('foo-abc-abc01', '192.168.1.11'),
             mlabconfig.format_a_record('foo-abc-mlab2-abc01', '192.168.1.24'),
-            mlabconfig.format_a_record('foo-abcv4-abc01', '192.168.1.11'),
             mlabconfig.format_a_record('foo-abc-mlab2v4-abc01', '192.168.1.24'),
-            mlabconfig.format_aaaa_record('foo-abc-abc01',
+            mlabconfig.format_aaaa_record('foo-abc-mlab3-abc01',
+                                          '2400:1002:4008::37'),
+            mlabconfig.format_aaaa_record('foo-abc-mlab1v6-abc01',
                                           '2400:1002:4008::11'),
+        ]
+
+        unexpected_results = [
+            mlabconfig.format_a_record('foo-abc-abc01', '192.168.1.24'),
+            mlabconfig.format_a_record('foo-abcv4-abc01', '192.168.1.24'),
             mlabconfig.format_aaaa_record('foo-abc-abc01',
                                           '2400:1002:4008::37'),
-            mlabconfig.format_aaaa_record('foo-abc-mlab3-abc01',
-                                          '2400:1002:4008::37'),
             mlabconfig.format_aaaa_record('foo-abcv6-abc01',
                                           '2400:1002:4008::11'),
-            mlabconfig.format_aaaa_record('foo-abc-mlab1v6-abc01',
-                                          '2400:1002:4008::11'),
         ]
 
         mlabconfig.SSL_EXPERIMENTS = ['abc_foo']
         mlabconfig.export_experiment_records(output, self.sites, experiments)
 
         results = output.getvalue().split('\n')
         self.assertContainsItems(results, expected_results)
+        self.assertDoesNotContainsItems(results, unexpected_results)
```

Comment:
It's a good instinct to verify that we are not generating records that we do not want.

And, I don't remember why I originally created `assertContainsItems` -- maybe `assertItemsEqual` was too fragile?

So, calling myself out here, I think this may be an example of the original code design making testing more difficult. And, there's no comment describing why we needed the approach with `assertContainsItems`.

When I'm looking now, it seems preferable to be able to make a single `assertItemsEqual` assertion to cover the two cases "has what we expect" and "doesn't have what I don't expect."

If `assertItemsEqual` does not work because of the comments injected into the ouput, then I can imagine it might be preferable to strip comments from `results` before comparing them. (Please do this if so).

If `assertItemsEqual` does not work because there are waaay too many records generated that are unrelated to the test, then... maybe having the two cases "contains" and "does not contain" is right. (Please (as a favor to me the next time I read this code) add the comment that I didn't add originally about why we need `assertContainsItems` and `assertDoesNotContainsItems`)